runtime/lib/backend/null_qubit/NullQubit.hpp

```diff
@@ -19,15 +19,44 @@
 #include <memory>
 #include <optional>
 #include <random>
+#include <unordered_map>
 #include <vector>
 
 #include "DataView.hpp"
 #include "QuantumDevice.hpp"
 #include "QubitManager.hpp"
 #include "Types.h"
 
+// The delimited to print before and after the JSON representing resource usage.
+//   This is intended to make parsing the output easier.
+//   Ideally, at some point resource usage would be passed back through as an object, and printing
+//   at all would be unneeded.
+#define RESOURCE_PRINT_DELIMETER "=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-="
+
 namespace Catalyst::Runtime::Devices {
 
+// TODO: For PR reviewers: Is this a good place to put this function?
+// It is used to print the resource usage of the device in a pretty format.
+// It is not used anywhere else in the codebase, but it feels like more of a utils type function
+// than anything that has to do with the device itself.
+template <class K, class V>
+void prettyPrint(const std::unordered_map<K, V> &map, size_t leadingSpaces = 0)
+{
+    std::cout << "{\n";
+    auto maplen = map.size();
+    for (const auto &pair : map) {
```

Comment:
You can use C++17 structured bindings to iterate over this map
```
for (const auto& [gate, size] : map) {
```

Reply:
Done in d7a2e25